server/rest/publish.py

```diff
@@ -48,16 +48,21 @@ def __init__(self):
             'token.',
             required=True,
         )
+        .param(
+            'isProduction',
+            description='Flag set to True when publishing to a production server',
+            required=True,
+        )
```

Comment:
I don't know the purpose of different D1 deployments, but it strikes me that it'd be more useful if we added an enum parameter that would accept any of `list(d1_common.env.D1_ENV_DICT.keys())`

Reply:
Each deployment has a different purpose. The development CN looks after a number of development member nodes. When testing new features this network is typically used. In our case, we're using to it test publishing (we push data to a dev member node, which the CN picks up). It also gives the user a chance to see what their package would look like, without actually putting it on production.

I like your idea of taking the keys as input, that would allow us to use _any_ CN specified in there. I'll go ahead and make that change.